chapter-outline.md

```diff
@@ -0,0 +1,287 @@
+      than drawing concepts maps,
+      but this would still ask them to do some retrieval as part of the drawing,
+      and there are plenty of retrieval exercises throughout the rest of the chapter.
+        - Again here it would be really cool to collect what students draw
+          and get ideas for how to improve the content.
+          - In general I think student do so much work and have so many ideas
+            that we are not yet good at leveraging as educators.
+            This is why I started the extra credit assignment in 531,
+            but more could be done I think
+            (although an easy solution would be to ask student to submit
+            what they drew as part of the extra credit
+            for a chapter where they thought the solution was missing something
+            that they had included).
+10. A challenging section where students who are interested and capable of going deeper
+    gets a chance to do so.
+    I open to that this section does not use the same data as the rest of the chapter.
+    That would allow us to include well known visualizations here,
+    and discuss some historical anecdotes
+    together with an advanced theoretical and/or coding section.
+    These sections can be brief and include just one (or even no?) exercise.
+    It's really to show interested students more depth
+    and pique their curiosity and "aw"-factor further.
+      - I have been pushing for that we should include these sections
+        in our courses in general in MDS,
+        so this would be a great opportunity to try out the concept.
+
+## Thoughts and questions
+
+- We want chapters to have a familiar structure to students
+  so that they know what to expect as they progress in the book,
+  but we don't want them to get bored.
+    - One thing I wonder is if there is value in that the chapter design evolves over time
+      akin to how levels in video games don't follow quite the same pattern
+      even if there are some similar elements like boss fights etc.
+      Is there value in introducing this novelty to make students stay engaged?
+      I think learning apps like Duolingo also have slightly different level design
+      although not as extensive as in a good video game.
+        - I'm curious about this,
+          but it is an "out there" idea I would say
+          and we would be on uncharted waters
+          if we start to evolve the book structure
+          as the story progresses.
+          I think it sounds fun,
+          and someone must have written something about this idea in the past
+          so maybe we can find a few academic articles to learn more
+          if it is considered an effective learning strategy.
+            - Maybe we can spend a little time looking at the idea,
+              but I don't want this to be a time sink up front
+              (although the more I think about it the more it enthralls me =p).
```

Comment:
Obviously I really like this idea. I wonder if there is a danger of making things addictive or engaging without actually enhancing the learning. Maybe if the gamification is intrinsically tied to gaining a deeper understanding.

Having said that, here are some gamification ideas:

- Use escalating narrative throughout the book. The stakes get higher. Maybe the student is aware that they will have to do something very difficult at the end.
- Skill trees: different chapters / sections could fill in a skill tree that expands as the student progresses through the book. Similar to Duolingo skill tree. It could fill in as they complete exercises.
- We could frame exercises as Challenges or something more game like.
- Perhaps more challenging exercises start to appear later in the book and you get more points for them
- If you do have a server setup you can consider info like x % of students completed this question or tell students what percentile they are in in terms of the number of points they've gained from completing challenges.

Reply:
>  I wonder if there is a danger of making things addictive or engaging without actually enhancing the learning. Maybe if the gamification is intrinsically tied to gaining a deeper understanding.

Good point that we should have in the back of our heads when designing.

---

Maybe some novel elements OK later as students get familiar with the book, but not complete chapter revamp since that would be confusing.

Maybe present LOs like a skill tree? Then students would  be able to overview what is in the course and how they make progress as they complete exercises.

Some indication of which exercises are more challenging (points? label?)

---

Server would be really nice from tracking perspective to be able to demonstrate the impact on the book and how students use it. Plausible.io/gogole analytics is another option here but they are not as granular